File: domain/src/main/java/org/openmedstack/domain/Saga.java
package org.openmedstack.domain;

public interface Saga {
    String getId();

    Integer getVersion();

    void transition(Object message) throws HandlerForDomainEventNotFoundException;

    Iterable<Object> getUncommittedEvents();

    void clearUncommittedEvents();

    Iterable<Object> getUndispatchedMessages();

    void clearUndispatchedMessages();
}

File: domain/src/main/kotlin/org/openmedstack/domain/Saga.kt
package org.openmedstack.domain

interface Saga {
    val id: String
    val version: Int
    @Throws(HandlerForDomainEventNotFoundException::class)
    fun transition(message: Any)
    val uncommittedEvents: Iterable<Any>
    fun clearUncommittedEvents()
    val undispatchedMessages: Iterable<Any>
    fun clearUndispatchedMessages()
}
